apps/test/unit/code-studio/pd/workshop_dashboard/workshopFactory.js

```diff
@@ -33,14 +38,74 @@ Factory.define('csp summer workshop')
   .attr('scholarship_workshop?', true)
   .attr('location_name', 'physical');
 
+Factory.define('csp summer workshop starting within month')
+  .extend('csp summer workshop')
+  .attr('sessions', () =>
+    Factory.buildList('session starting within month', 2)
+  );
+
+Factory.define('csd summer workshop starting within month')
+  .extend('csp summer workshop starting within month')
+  .attr('course', 'CS Discoveries');
+
+Factory.define('csp summer workshop starting in over a month')
+  .extend('csp summer workshop')
+  .attr('sessions', () =>
+    Factory.buildList('session starting in over a month', 1)
+  );
+
+/**
+ * SESSIONS
+ */
 Factory.define('session')
   .sequence('id')
   .attr('code', 'TEST')
-  .attr('start', new Date().toISOString())
-  .attr('end', new Date().toISOString())
+  .attr('start', today.toISOString())
+  .attr('end', today.toISOString())
   .attr('attendance_count', 0)
   .attr('show_link?', false);
 
+Factory.define('session starting within month')
+  .extend('session')
+  .attr(
+    'start',
+    new Date(
+      today.getFullYear(),
+      today.getMonth(),
+      today.getDate() + 15
+    ).toISOString()
+  )
+  .attr(
+    'end',
+    new Date(
+      today.getFullYear(),
+      today.getMonth(),
+      today.getDate() + 16
+    ).toISOString()
+  );
+
+Factory.define('session starting in over a month')
+  .extend('session')
+  .attr(
+    'start',
+    new Date(
+      today.getFullYear(),
+      today.getMonth() + 2,
+      today.getDate()
+    ).toISOString()
+  )
+  .attr(
+    'end',
+    new Date(
+      today.getFullYear(),
+      today.getMonth() + 2,
+      today.getDate()
+    ).toISOString()
+  );
```

Comment:
nice job adding these tests! one concern about using `today` is that it can cause our tests to behave differently depending on the current date/time, which is something we want to avoid. could you please change these to us a specific date of your choice, and then also include tests showing that the "over a month" logic still works across the end-of-year boundary?

Reply:
Since it has to check if the workshop is within a month, the logic uses `today`, is it possible for me to set an arbitrary "today" like with TimeCop? And yes I can also add that!

Reply:
ah I see, this is tricky! here are a couple ideas:
1. extract a helper method like [this one](https://github.com/code-dot-org/code-dot-org/blob/54f5a314e5c812e03a008afed2127d6080601f04/apps/src/utils.js#L739-L741) to fetch the current date, which you can then stub out in your unit tests (like [here](https://github.com/code-dot-org/code-dot-org/blob/563e25389306aa2f0ef2dd8e72c7198da73edb1d/apps/test/unit/code-studio/initApp/projectTest.js#L743))
2. do the date calculation on the server. this is a slightly bigger change, but will be more reliable since browser date/time is not reliable anyway.

if you are feeling ambitious then (2) would be best, but since you've already got most of this working and (per other discussion topic) the quality bar seems to be "best effort" right now, then I think (1) is also fine and could avoid undoing a lot of your existing work and testing.

one thing to call out is that if we do go with (1), there's the risk that someone with a wrong time on their machine won't be able to edit the "virtual" field on their workshop even if it is more than a month out, but since by definition they'll have over a month before their workshop in that scenario it seems like this risk is unlikely to be super disruptive (i.e. they will have time to contact us well ahead of their workshop).

Reply:
I'm down to look into option 2! Would you mind explaining what you mean by "do the date calculation on the server" please? Like would that entail obtaining the date on the ruby side and passing that over to the workshop form to use there? Or am I misunderstanding this?

Reply:
sorry to leave out important details! at the very least, today's date should be be calculated on the server. but unless you need it for anything else, I would recommend calculating `workshopStartsWithinMonth` on the server, and passing that down as a boolean, in which case you won't need to pass down today's date. that will let you keep some of the logic you have on the client.

Reply:
No worries at all! That's what I originally had but had to switch since when a user is creating a workshop, the workshop db entry isn't added until they submit the form so I wouldn't be able to access the dates of the workshop on the backend to check if they're starting within a month. That's why I moved it all to the front-end so that I can access the current values they have entered, plus this unifies the behavior regardless of if they're creating a new workshop or editing an existing one. So I'm thinking maybe I'll just calculate today's date on teh server-side and pass that in but keep the front-end logic? Curious on your thoughts on that!